src/arch/arm/object/vcpu.c

```diff
@@ -280,6 +280,10 @@ void associateVCPUTCB(vcpu_t *vcpu, tcb_t *tcb)
     }
     tcb->tcbArch.tcbVCPU = vcpu;
     vcpu->vcpuTCB = tcb;
+
+    if (tcb == NODE_STATE(ksCurThread)) {
```

Comment:
Yes, that is correct. The intended scenario is for when the currently running thread binds itself to a VCPU, regardless of whether it was previously bound to one or not. If it was previously not bound to one, then my understanding is that this can be considered as the thread promoting itself to a VM.

Reply:
With regards to the function name, I just thought of the option of doing the `vcpu_switch` in `invokeVCPUSetTCB`, just after calling `associateVCPUTCB`. Would this make more sense? It would require slightly more effort on the proof side, but I don't think that it would be much more.

@michaelmcinerney do you agree that the proof of the invariant could handle moving the `vcpu_switch`?

Reply:
The proofs for the invariant itself should be quite straightforward. I imagine it would be much more annoying to do the `invs/invs'` proofs and `corres/ccorres` proofs within the invocation function

Reply:
Given that the function is called only precisely once (in`invokeSetTCB`), I'd vote for changing the name to indicate that it also might switch instead of pulling functionality out that is necessary for preserving invariants.